src/api/cvc4cpp.cpp

```diff
@@ -1204,5 +1204,233 @@ std::ostream& operator<<(std::ostream& out,
+  // == NodeManager::fromExprManager(child2.getExprManager())
+  // CHECK:
+  // const Metakind mk = kind::metaKindOf(kind);
+  // mk != kind::metakind::PARAMETERIZED && mk != kind::metakind::OPERATOR
+  // else "Only operator-style expressions are made with mkExpr(); "
+  //      "to make variables and constants, see mkVar(), mkBoundVar(), "
+  //      "and mkConst()."
+  // CHECK:
+  // const unsigned n = 2 - (mk == kind::metakind::PARAMETERIZED ? 1 : 0);
+  // n < minArity(kind) || n > maxArity(kind)
+  // else "Exprs with kind %s must have at least %u children and "
+  //      "at most %u children (the one under construction has %u)"
+  return d_exprMgr->mkExpr(extToIntKind(kind), *child1.d_expr, *child2.d_expr);
+}
+
+Term Solver::mkTerm(Kind kind, Term child1, Term child2, Term child3) const
+{
+  // CHECK:
+  // NodeManager::fromExprManager(d_exprMgr)
+  // == NodeManager::fromExprManager(child1.getExprManager())
+  // NodeManager::fromExprManager(d_exprMgr)
+  // == NodeManager::fromExprManager(child2.getExprManager())
+  // NodeManager::fromExprManager(d_exprMgr)
+  // == NodeManager::fromExprManager(child3.getExprManager())
+  // CHECK:
+  // const Metakind mk = kind::metaKindOf(kind);
+  // mk != kind::metakind::PARAMETERIZED && mk != kind::metakind::OPERATOR
+  // else "Only operator-style expressions are made with mkExpr(); "
+  //      "to make variables and constants, see mkVar(), mkBoundVar(), "
+  //      "and mkConst()."
+  // CHECK:
+  // const unsigned n = 3 - (mk == kind::metakind::PARAMETERIZED ? 1 : 0);
+  // n < minArity(kind) || n > maxArity(kind)
+  // else "Exprs with kind %s must have at least %u children and "
+  //      "at most %u children (the one under construction has %u)"
+  std::vector<Expr> echildren{*child1.d_expr, *child2.d_expr, *child3.d_expr};
+  CVC4::Kind k = extToIntKind(kind);
+  return kind::isAssociative(k) ? d_exprMgr->mkAssociative(k, echildren)
+                                : d_exprMgr->mkExpr(k, echildren);
+}
+
+Term Solver::mkTerm(Kind kind, const std::vector<Term>& children) const
+{
+  // CHECK:
+  // for c in children:
+  // NodeManager::fromExprManager(d_exprMgr)
+  // == NodeManager::fromExprManager(c.getExprManager())
+  // CHECK:
+  // const Metakind mk = kind::metaKindOf(kind);
+  // mk != kind::metakind::PARAMETERIZED && mk != kind::metakind::OPERATOR
+  // else "Only operator-style expressions are made with mkExpr(); "
+  //      "to make variables and constants, see mkVar(), mkBoundVar(), "
+  //      "and mkConst()."
+  // CHECK:
+  // const unsigned n = children.size() - (mk == kind::metakind::PARAMETERIZED ?
+  // 1 : 0); n < minArity(kind) || n > maxArity(kind) else "Exprs with kind %s
+  // must have at least %u children and "
+  //      "at most %u children (the one under construction has %u)"
+  std::vector<Expr> echildren = termVectorToExprs(children);
+  CVC4::Kind k = extToIntKind(kind);
+  return kind::isAssociative(k) ? d_exprMgr->mkAssociative(k, echildren)
```

Comment:
`isAssociative` is currently incomplete: https://github.com/CVC4/CVC4/blob/14b9dbaa0c9e8dce52d1a28595dc1cc80756abed/src/expr/kind_template.cpp#L39-L54

Are there plans to address this? I guess it is not super critical since `mkAssociative` just seems to break up operators with lots of children but we should have a plan going forward at least.